tests/cli/test_utils.py

```diff
@@ -403,6 +405,66 @@ def _job_entry(job_id: str) -> Dict[str, Any]:
             await resolve_job(uri, client=client, default_user="default-owner")
 
 
+async def test_parse_and_resolve_resource_for_sharing_job_name_with_owner(
+    aiohttp_server: _TestServerFactory, make_client: _MakeClient
+) -> None:
+    job_name = "job-name"
+    owner_name = "owner-name"
+    uri = f"job://{owner_name}/{job_name}"
+    job_id = "job-id-1"
+    JSON = {"jobs": [_job_entry(job_id)]}
+
+    async def handler(request: web.Request) -> web.Response:
+        # Since `resolve_job` excepts any Exception, `assert` will be caught there
+        name = request.query.get("name")
+        if name != job_name:
+            pytest.fail(f"received: {name}")
```

Comment:
```
            pytest.fail(f"received name: {name}")
```
or
```
            pytest.fail(f"received: {request.query}")
```
or even
```
        assert request.query.get("name") == job_name
```

Reply:
> assert request.query.get("name") == job_name

this doesn't work as this assertion error will be caught by try-catch block within `resolve_job()`, so the result of test will be silently different. I agree, `except Exception` sounds like a bad design, but for this method, which follows the strategy "try to resolve, if *anything* goes wrong, roll back to the original argument silently", it's fine IMO.

Reply:
fixed to 
```
pytest.fail(f"received name: {name}")
```
thanks